packages/mobile/src/set-clock/SetClock.tsx

```diff
@@ -8,17 +8,28 @@ import { Image, Platform, StyleSheet, Text, View } from 'react-native'
 import * as AndroidOpenSettings from 'react-native-android-open-settings'
 import { Namespaces, withTranslation } from 'src/i18n'
 import { clockIcon } from 'src/images/Images'
-import { navigateHome } from 'src/navigator/NavigationService'
+import { restartApp } from 'src/utils/AppRestart'
 import { getLocalTimezone, getRemoteTime } from 'src/utils/time'
 
-export class SetClock extends React.Component<WithTranslation> {
-  static navigationOptions = { header: null }
+interface State {
+  correctTime: number
+}
+
+export class SetClock extends React.Component<WithTranslation, State> {
+  state = {
+    correctTime: Date.now(),
+  }
+
+  componentDidMount = async () => {
+    const correctTime = await getRemoteTime()
+    this.setState({ correctTime })
+  }
 
-  goToSettings = () => {
+  goToSettingsOrRestart = () => {
     if (Platform.OS === 'android') {
       return AndroidOpenSettings.dateSettings()
     } else {
-      navigateHome()
+      restartApp()
```

Comment:
I think restartApp might not work on iOS from what I remember with the forno toggle

Reply:
Yeah, it will just close the app. The button says "Quit" rather than "Restart" for that reason